File: src/AdventureWorks.Application/DataEngine/Production/ProductCostHistory/Queries/GetProductCostHistories/ProductCostHistoryLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Production;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;


namespace AdventureWorks.Application.DataEngine.Production.ProductCostHistory.Queries.GetProductCostHistories
{
    public partial class ProductCostHistoryLookupModel : IHaveCustomMapping
    {
        public int ProductID { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public decimal StandardCost { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Production.ProductCostHistory, ProductCostHistoryLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<ProductCostHistoryLookupModel, BaseProductCostHistory>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/production/product-cost-history/product-cost-history-lookup-model.ts

export interface IProductCostHistoryLookupModel {
  productID: number;
  startDate: Date | string;
  endDate: Date | string | null;
  standardCost: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class ProductCostHistoryLookupModel implements IProductCostHistoryLookupModel {
  productID: number;
  startDate: Date | string;
  endDate: Date | string | null;
  standardCost: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<ProductCostHistoryLookupModel>) {
    if (!!init) {
      Object.assign<ProductCostHistoryLookupModel, Partial<ProductCostHistoryLookupModel>>(this, init);
    }
  }
}
